tools/omero/omero_basic_upload/omero_import.xml

```diff
@@ -0,0 +1,72 @@
+<tool id="omero_import" name="OMERO Image Import" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="20.01" license="MIT">
+    <description> with omero-py </description>
+    <macros>
+        <token name="@TOOL_VERSION@">5.18.0</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+    </macros>
+    <xrefs>
+        <xref type="bio.tools">omero</xref>
+    </xrefs>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">omero-py</requirement>
+        <requirement type="package" version="21.0.2">openjdk</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        mkdir folder &&
+        #for $image in $folder:
+            ln -s '$image' folder/'$image.element_identifier' &&
```

Comment:
Since the `element_identifier` may contain characters that do not work on the command line  (i.e. are not allowed to be used / used safely in filenames) we need to do some regex search replace here. Typically something like [this](https://github.com/galaxyproject/tools-iuc/blob/9af8730b86781bcbbe1cb62402967fa316cfec80/tools/vsnp/vsnp_statistics.xml#L11) is done.

Reply:
Is the file extension used at all?

It would be fine to add `.` if you assume that the extension is stored in the identifier. Otherwise you may use `$image.ext` which is the name of the Galaxy datatype of the dataset.

Reply:
what about:
```
 #for $image in $folder:
            #set image = $image
            #set image_identifier = re.sub('[^\s\w\-]', '_', str($image.element_identifier))
            ln -s '${image}' folder/'${image_identifier}.$image.ext' &&
        #end for
```

Reply:
looks good to me.